packages/pigeon/lib/ast.dart

```diff
@@ -83,9 +83,18 @@ class TypeDeclaration {
     this.typeArguments = const <TypeDeclaration>[],
   });
```

Comment:
Optional: Maybe assert that `baseName` isn't `void`, to ensure nobody accidentally creates, say, `TypeDeclaration(baseName: 'void', isNullable: true)`, and instead uses `voidDeclaration`?

(I know there are other invalid things someone could do, like trying to make `int<double>`, but since void is a special enough case that it's getting its own helpers, so it seems like it might make sense to special-case here.)

Reply:
I wish I could make it a static check.  I'm worried I could accidentally introduce a runtime exception when I incorrectly rebase outstanding PRs.